pkg/dnfjson/dnfjson.go

```diff
@@ -196,7 +196,7 @@ func (s *Solver) SetProxy(proxy string) error {
 func (s *Solver) Depsolve(pkgSets []rpmmd.PackageSet) ([]rpmmd.PackageSpec, []rpmmd.RepoConfig, error) {
 	req, rhsmMap, err := s.makeDepsolveRequest(pkgSets)
 	if err != nil {
-		return nil, nil, err
+		return nil, nil, fmt.Errorf("makeDepsolveRequest failed: %w", err)
```

Comment:
I tend to forget our minimal Go version requirement but if it's `>= 1.20` I would prefer `errors.Join(errors.New("makeDepsolveRequest failed"), err)` here.

Reply:
I'm curious why `errors.Join()` is preferred here. My understanding so far was that `%w` is used to provide additional context (which seems to be the case here) and errors.Join() is more common in cases when handling one error results in an additional erorr (the classic example would be an error from `defer os.Close()`). But maybe I'm missing something here or am not fully up-to-date?

Reply:
Both can do both (starting at 1.20) historically `%w` could only be used once so that probably shoehorned it into the additional context role. It's a preference, I find it to be a bit easier to read (like a slice of errors).

Reply:
> I tend to forget our minimal supported Go version requirement but if it's `>= 1.20` I would prefer `errors.Join(errors.New("makeDepsolveRequest failed"), err)` for this idiom.

who can confirm our minimal supported golang version?

Reply:
The go version according to go.mod is 1.20. And yeah, of course %w and errors.Join can do the same, I was mostly wondering what the idiomatic way is. Looking at the go standard library it seems `errors.Join()` is mostly used when there are multiple errors from e.g. multiple go routines or multiple files (e.g. modload/init.go) and it's relatively rare (I counted 6 with a quick grep in go-1.22). And %w is used a lot more (135) and it seems (from a subset of samples I looked at) that it is mostly used to provide extra context (I did not find this for errors.Join())). Based on this it seems the use in this PR is fine (but of course there is an argument that errors.Join is relatively new and the usage has just not caught up yet in the stdlib).